internal/flushcommon/writebuffer/stats_buffer.go

```diff
@@ -0,0 +1,49 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writebuffer
+
+import (
+	"github.com/milvus-io/milvus/internal/storage"
+)
+
+type statsBuffer struct {
+	meta map[int64]*storage.BM25Stats
+}
+
+func (b *statsBuffer) Buffer(metaMap map[int64]*storage.BM25Stats) {
+	for fieldID, meta := range metaMap {
```

Comment:
check for `b == nil`, b may be nil if there is no function in collection schema, and seg fault will happen in such cases